### PR TITLE
Currency: Require space after unit abbreviation

### LINKED ISSUE #2533: Currency: triggers on "mars" because of `m` prefix + `ars` symbol
It looks like we have a regex that checks for `million` but also `m`. As well, `ars` is a currency name.

So together, `mars` looks like a valid trigger. This also happens for `musd` for example.

I think if the trigger contains `million` or `m` it should also have a number, or we may want to whitelist the phrase `mars`.

@laouji @MrChrisW, what are your thoughts?

------
IA Page:  http://duck.co/ia/view/currency

### PR DESCRIPTION
**Pull request by MrChrisW**
Fixes #2533 

//CC @moollaza 

------
IA Page:  http://duck.co/ia/view/currency

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by MrChrisW** [comment]
Bump! @moollaza

File: lib/DDG/Spice/Currency.pm [hunk 1]
```diff
@@ -28,7 +28,8 @@ my $into_qr = qr/\s(?:en|in|to|in ?to|to)\s/i;
 my $vs_qr = qr/\sv(?:ersu|)s\.?\s/i;
 my $question_prefix = qr/(?:convert|what (?:is|are|does)|how (?:much|many) (?:is|are))?\s?/;
 my $number_re = number_style_regex();
-my $cardinal_re = join('|', qw(hundred thousand k million m billion b trillion));
+my $cardinal_re = qr/hundred |thousand |k |million |m |billion |b |trillion /;
```
Comment (GuiltyDolphin) [inline lib/DDG/Spice/Currency.pm]: Couldn't we just use `join('| ', qw(...))`?

File: t/Currency.t [hunk 2]
```diff
@@ -107,7 +107,7 @@ ddg_spice_test(
         is_cached => 0
     ),
     # Query with everything smushed together, with k for thousand.
-    '2kcadusd' => test_spice(
+    '2k cadusd' => test_spice(
```
Comment (GuiltyDolphin) [inline t/Currency.t]: Can we add a couple more valid tests for the new behavior?

File: lib/DDG/Spice/Currency.pm (hunk 1 above)
Comment (MrChrisW) [inline lib/DDG/Spice/Currency.pm]: Nope, that will result in `hundred| thousand| k| million| m| billion| b| trillion` not `hundred |thousand |k |million |m |billion |b |trillion`

Comment (GuiltyDolphin) [inline lib/DDG/Spice/Currency.pm]: Whoops, my bad!

Comment (MrChrisW) [inline lib/DDG/Spice/Currency.pm]: @GuiltyDolphin haha I just made an edit suggesting that.. I'm testing now.

File: lib/DDG/Spice/Currency.pm [hunk 3]
```diff
@@ -124,15 +125,18 @@ handle query_lc => sub {
...
-            if ($cardinal eq 'hundred')  { $amount *= 100 }
-            elsif ($cardinal =~ /(thousand|k)/i) { $amount *= 1000 }
-            elsif ($cardinal =~ /(million|m)/i)  { $amount *= 1000000 }
-            elsif ($cardinal =~ /(billion|b)/i)  { $amount *= 1000000000 }
-            elsif ($cardinal =~ /(trillion|t)/i) { $amount *= 1000000000000 }
+            if ($cardinal =~ /(hundred )/i)  { $amount *= 100 }
+            elsif ($cardinal =~ /(thousand |k )/i) { $amount *= 1000 }
+            elsif ($cardinal =~ /(million |m )/i)  { $amount *= 1000000 }
+            elsif ($cardinal =~ /(billion |b )/i)  { $amount *= 1000000000 }
+            elsif ($cardinal =~ /(trillion |t )/i) { $amount *= 1000000000000 }
```
Comment (GuiltyDolphin) [inline lib/DDG/Spice/Currency.pm]: Can we format these with underscores (make them easier to read); e.g., `1_000_000`.

**Comment by MrChrisW** [comment]
@GuiltyDolphin Thanks :+1:

File: lib/DDG/Spice/Currency.pm (hunk 3 above)
Comment (MrChrisW) [inline lib/DDG/Spice/Currency.pm]: Great suggestion :smile:

File: lib/DDG/Spice/Currency.pm [hunk 4]
```diff
@@ -28,7 +28,8 @@ my $into_qr = qr/\s(?:en|in|to|in ?to|to)\s/i;
 my $vs_qr = qr/\sv(?:ersu|)s\.?\s/i;
 my $question_prefix = qr/(?:convert|what (?:is|are|does)|how (?:much|many) (?:is|are))?\s?/;
 my $number_re = number_style_regex();
-my $cardinal_re = join('|', qw(hundred thousand k million m billion b trillion));
+my $cardinal_re = join(' |', qw(hundred thousand k million m billion b trillion));
```
Comment (GuiltyDolphin) [inline lib/DDG/Spice/Currency.pm]: I think we'll need to use `join(...) . ' '` to get a space after `trillion`.

Comment (MrChrisW) [inline lib/DDG/Spice/Currency.pm]: @GuiltyDolphin Now it's ugly again... :cry:

Comment (GuiltyDolphin) [inline lib/DDG/Spice/Currency.pm]: Well the alternative would be `@{[join(' |', qw(...))]} ` :wink:

**Comment by GuiltyDolphin** [comment]
@MrChrisW Okay, thanks for this! :dolphin:

File: lib/DDG/Spice/Currency.pm [hunk 5]
```diff
@@ -28,7 +28,8 @@ my $into_qr = qr/\s(?:en|in|to|in ?to|to)\s/i;
 my $vs_qr = qr/\sv(?:ersu|)s\.?\s/i;
 my $question_prefix = qr/(?:convert|what (?:is|are|does)|how (?:much|many) (?:is|are))?\s?/;
 my $number_re = number_style_regex();
-my $cardinal_re = join('|', qw(hundred thousand k million m billion b trillion));
+my $cardinal_re = join(' |', qw(hundred thousand k million m billion b trillion)).' ';
+
```
Comment (MrChrisW) [inline lib/DDG/Spice/Currency.pm]: @GuiltyDolphin Nooo additional new line! Oh well :+1:

Comment (GuiltyDolphin) [inline lib/DDG/Spice/Currency.pm]: @MrChrisW I wouldn't lose too much sleep over it :wink:

**Pull request merged**